### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by agaengel**

### CODE REVIEW

File: lib/configuration_parser.rb [hunk 1]
```diff
@@ -21,13 +25,18 @@ def parse(options)
 
     def parse_1_0(config, dist_folder, bootstrap)
       if config.include? 'deploy'
+        abort "only 'bootstrap' or 'recurring are allowed for deploy.fore in .deploy-now/config.yaml" unless config['deploy']['force'].eql?('bootstrap') || config['deploy']['force'].eql?('recurring') || config['deploy']['force'].nil?
```
Comment (coder-hugo) [inline lib/configuration_parser.rb:28]: There is a typo. Should be `deploy.force` not `deploy.fore`

File: lib/size_checker.rb [hunk 2]
```diff
@@ -19,38 +24,42 @@ def check(options)
     private
 
     def calculate_local_size(dist_folder, excludes)
+      puts get_entries(dist_folder)
```
Comment (coder-hugo) [inline lib/size_checker.rb:27]: Was this `puts` for debugging purposes?

File: lib/size_checker.rb [hunk 3]
```diff
@@ -19,38 +24,42 @@ def check(options)
...
+
+      exclude_options = (excludes).map { |exclude| "--exclude=#{exclude}" }.join(' ')
 
       Net::SSH.start(host, user[:username], password: user[:password], verify_host_key: :never) do |ssh|
-        ssh.exec!('shopt -s dotglob; du -sb *')
-           .scan(/(\d+)\s+(\S+)/)
-           .select { |_, path| dirs.include? path }
-           .map { |size, _| size.to_i }
-           .sum
+        ssh.exec!("shopt -s dotglob; expr $(du -sb . | cut -f1 ) - $(du -sb . #{exclude_options} | cut -f1)").to_i
```
Comment (coder-hugo) [inline lib/size_checker.rb:62]: The `shopt -s dotglob` is obsolete with this change. This changes the behavior of the shell regarding an asterisk (`*`).

File: lib/configuration_parser.rb (hunk 1 above)
Comment (agaengel) [reply lib/configuration_parser.rb]: fixed

**Pull request merged**